beacon_node/beacon_chain/src/snapshot_cache.rs

```diff
@@ -253,7 +253,8 @@ impl<T: EthSpec> SnapshotCache<T> {
             .position(|snapshot| snapshot.beacon_block_root == block_root)
             .map(|i| {
                 if let Some(cache) = self.snapshots.get(i) {
-                    if block_slot > cache.beacon_block.slot() + 1 {
+                    // Avoid cloning the block during sync (when the `block_delay` is `None`).
+                    if block_slot > cache.beacon_block.slot() + 1 && block_delay.is_some() {
```

Comment:
Very very minor point in the case where `block_delay` is `None`:
If we nest this `if` block below into the `if let Some` block would it avoid us checking `block_delay` twice (assuming `block_slot > cache.beacon_block.slot() + 1 == true`)?

Reply:
yeah good call, that's simpler